ios/brave-ios/Sources/Brave/WebFilters/FilterListResourceDownloader.swift

```diff
@@ -13,35 +13,29 @@ import Shared
 import os.log
 
 /// An object responsible for handling filer lists changes on `FilterListStorage` and registering them with the `AdBlockService`.
-public actor FilterListResourceDownloader {
+@MainActor public class FilterListResourceDownloader {
   /// A shared instance of this class
   ///
   /// - Warning: You need to wait for `DataController.shared.initializeOnce()` to be called before using this instance
   public static let shared = FilterListResourceDownloader()
-
-  /// Object responsible for getting component updates
-  private var adBlockService: AdblockService?
-  /// Ad block service tasks per filter list UUID
-  private var adBlockServiceTasks: [String: Task<Void, Error>]
   /// A marker that says that we loaded shield components for the first time.
   /// This boolean is used to configure this downloader only once after `AdBlockService` generic shields have been loaded.
-  private var registeredFilterLists = false
+  private var subscribedToFilterListChange = false
+  /// The filter list change subscription
+  private var filterListSubscription: AnyCancellable?
+  /// This is a task that we use to delay a requested compile.
+  /// This allows us to wait for more sources and limit the number of times we compile
+  private var delayTasks: [GroupedAdBlockEngine.EngineType: Task<Void, Error>] = [:]
 
-  init() {
-    self.adBlockServiceTasks = [:]
-    self.adBlockService = nil
-  }
+  init() {}
 
   /// Start the adblock service to get updates to the `shieldsInstallPath`
   public func start(with adBlockService: AdblockService) {
-    self.adBlockService = adBlockService
-
     // Here we register components in a specific order in order to avoid race conditions:
     // 1. All our filter lists need the resources component. So we register this first
-    // 2. We also register the catalogue component since we don't need the resources for this
-    // 3. When either of the above triggers we check if we have our resources and catalogue
-    // and if we do have both we register all of the filter lists.
-    Task { @MainActor in
+    // 2. We start the custom filter list downloader
+    // 3. We register the filter list changes
+    Task {
```

Comment:
This will run on the inherited task (possibly main) depending on what task `start(with adBlockService: AdblockService)` runs on. Is this intentional? I ask because you removed `@MainActor`.

Reply:
The class is MainActor isolated (formally was actor isolated) so it should pick up that isolation (I really don't believe it picks up the Adblock service isolation here but maybe im wrong) so I didn't think it needs this specified any more.  I can keep it to be sure but I really don't think it's necessary anymore.